query-engine/query-engine/src/server/mod.rs

```diff
@@ -462,6 +356,37 @@ fn err_to_http_resp(err: query_core::CoreError) -> Response<Body> {
 
     let user_error: user_facing_errors::Error = err.into();
     let body = Body::from(serde_json::to_vec(&user_error).unwrap());
-
     Response::builder().status(status).body(body).unwrap()
 }
+
+fn traceparent(headers: &HeaderMap) -> Option<String> {
+    const TRACEPARENT_HEADER: &str = "traceparent";
+
+    let value = headers
+        .get(TRACEPARENT_HEADER)
+        .and_then(|h| h.to_str().ok())
+        .map(|s| s.to_owned());
+
+    let is_valid_traceparent = |s: &String| s.split_terminator('-').count() >= 4;
+
+    match &value {
+        Some(str) if is_valid_traceparent(str) => value,
+        _ => None,
+    }
```

Comment:
Some people really hate `Option::filter()` so don't feel forced to use it, but you can write that match statement as `value.filter(is_valid_traceparent)`

Reply:
Why do people hate it, seems to make sense.

Reply:
My guess is years of JS experience where your brain automatically goes "oh, this is a collection/array" when you read "filter", and so you're surprised that it's a method on an Option.

Reply:
OK, that's why I don't hate it lol